ethereum/core/src/main/java/tech/pegasys/pantheon/ethereum/mainnet/precompiles/privacy/PrivacyPrecompiledContract.java

```diff
@@ -89,18 +105,40 @@ public BytesValue compute(final BytesValue input, final MessageFrame messageFram
       PrivateTransaction privateTransaction = PrivateTransaction.readFrom(bytesValueRLPInput);
 
       WorldUpdater publicWorldState = messageFrame.getWorldState();
-      WorldUpdater privateWorldState = privateWorldStateArchive.getMutable().updater();
+      // get the last world state root hash - or create a new one
+      BytesValue privacyGroupId = BytesValue.wrap("0".getBytes(UTF_8));
+
+      Hash lastRootHash =
+          privateStateStorage.getPrivateAccountState(privacyGroupId).orElse(EMPTY_ROOT_HASH);
+      MutableWorldState disposablePrivateState =
+          privateWorldStateArchive.getMutable(lastRootHash).get();
+
+      WorldUpdater privateWorldStateUpdater = disposablePrivateState.updater();
+
       TransactionProcessor.Result result =
           privateTransactionProcessor.processTransaction(
               messageFrame.getBlockchain(),
               publicWorldState,
-              privateWorldState,
+              privateWorldStateUpdater,
               messageFrame.getBlockHeader(),
               privateTransaction,
               messageFrame.getMiningBeneficiary(),
               OperationTracer.NO_TRACING,
               messageFrame.getBlockHashLookup());
 
+      privateWorldStateUpdater.commit();
```

Comment:
Don't we need to check the result from `privateTransactionProcessor.processTransaction()` to know if the transaction has been processed successfully?